src/App.tsx

```diff
@@ -14,11 +14,109 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  selectedGoods: string[];
+};
+
+class App extends React.Component<{}, State> {
+  state: State = {
+    selectedGoods: ['Jam'],
+  };
+
+  clearGoods = () => {
+    this.setState({
+      selectedGoods: [],
+    });
+  };
+
+  selectGood = (good: string) => {
+    this.setState((prevState) => {
+      if (prevState.selectedGoods.includes(good)) {
+        const newSelectedGoods = prevState.selectedGoods
+          .filter(item => item !== good);
+
+        return {
+          selectedGoods: newSelectedGoods,
+        };
+      }
+
+      const newSelectedGoods = [
+        ...prevState.selectedGoods,
+        good,
+      ];
+
+      return {
+        selectedGoods: newSelectedGoods,
+      };
+    });
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+    let title = '';
+
+    switch (selectedGoods.length) {
+      case 0:
+        title = 'No goods selected';
+        break;
+
+      case 1:
+        title = `${selectedGoods[0]} is selected`;
+        break;
+
+      case 2:
+        title = `${selectedGoods[0]} and ${selectedGoods[1]} are selected`;
+        break;
+
+      default: {
+        const goodsWithoutLast = selectedGoods.slice(0, -1).join(',');
+        const lastGood = selectedGoods.slice(-1);
+
+        title = `${goodsWithoutLast} and ${lastGood} are selected`;
+        break;
+      }
```

Comment:
don't need add break for default option 

```suggestion
      default: {
        const goodsWithoutLast = selectedGoods.slice(0, -1).join(',');
        const lastGood = selectedGoods.slice(-1);

        title = `${goodsWithoutLast} and ${lastGood} are selected`;
      }
```